src/gui/folderwatcher_win.cpp

```diff
@@ -97,47 +97,56 @@ void WatcherThread::watchChanges(size_t fileNotifyBufferSize,
+            qCWarning(lcFolderWatcher) << "WaitForMultipleObjects failed" << result << Utility::formatWinError();
             break;
         }
 
         bool ok = GetOverlappedResult(_directory, &overlapped, &dwBytesReturned, false);
         if (!ok) {
-            DWORD errorCode = GetLastError();
+            const DWORD errorCode = GetLastError();
             if (errorCode == ERROR_NOTIFY_ENUM_DIR) {
                 qCDebug(lcFolderWatcher) << "The buffer for changes overflowed! Triggering a generic change and resizing";
                 emit lostChanges();
                 emit changed(_path);
                 *increaseBufferSize = true;
             } else {
-                qCWarning(lcFolderWatcher) << "GetOverlappedResult error" << errorCode;
+                qCWarning(lcFolderWatcher) << "GetOverlappedResult error" << Utility::formatWinError(errorCode);
             }
             break;
         }
 
         FILE_NOTIFY_INFORMATION *curEntry = pFileNotifyBuffer;
         forever {
-            size_t len = curEntry->FileNameLength / 2;
-            QString file = _path + "\\" + QString::fromWCharArray(curEntry->FileName, len);
+            const int len = curEntry->FileNameLength / sizeof(wchar_t);
+            QString longfile = longPath + QString::fromWCharArray(curEntry->FileName, len);
 
             // Unless the file was removed or renamed, get its full long name
             // TODO: We could still try expanding the path in the tricky cases...
-            QString longfile = file;
             if (curEntry->Action != FILE_ACTION_REMOVED
                 && curEntry->Action != FILE_ACTION_RENAMED_OLD_NAME) {
-                size_t longNameSize = GetLongPathNameW(reinterpret_cast<LPCWSTR>(file.utf16()), fileNameBuffer, fileNameBufferSize);
+                const auto wfile = longfile.toStdWString();
+                const int longNameSize = GetLongPathNameW(wfile.data(), fileNameBuffer, fileNameBufferSize);
                 if (longNameSize > 0) {
-                    longfile = QString::fromUtf16(reinterpret_cast<const ushort *>(fileNameBuffer), longNameSize);
+                    longfile = QString::fromWCharArray(fileNameBuffer, longNameSize);
                 } else {
-                    qCWarning(lcFolderWatcher) << "Error converting file name to full length, keeping original name.";
+                    qCWarning(lcFolderWatcher) << "Error converting file name" << longfile << "to full length, keeping original name." << Utility::formatWinError();
                 }
             }
+
+#if QT_VERSION < QT_VERSION_CHECK(5, 14, 0)
+            // The prefix is needed for native Windows functions before Windows 10, version 1607
+            const bool hasLongPathPrefix = longPath.startsWith(QStringLiteral("\\\\?\\"));
+            if (hasLongPathPrefix)
+            {
+                longfile.remove(0, 4);
+            }
+#endif
```

Comment:
Any more info on this?

Reply:
I changed the longFile name to start with longPath which is a windows file path including the \\\\?\\ prefix which was needed on windows <10 to allow paths longer than 260 chars.

Reply:
I rather meant the qt_version_check, what changed when and how etc. If you know more, could be added as comments.

Reply:
I had to dive into the code to find out why the unit tests where failing and why it was working with my Qt 5.15 setup and not on the 5.12.9 ci...

Reply:
https://github.com/qt/qtbase/commit/3966b571ff54100923a479050a38ac906e0ed1e3